src/main/java/useCase/DatabaseManager.java

```diff
@@ -23,162 +21,172 @@ public static DatabaseManager getDatabaseManager() {
+            for (String blocked_by_users : blocked_by_users_list) {
+                UserAccount blockedByUser = retrieveUserAccount(blocked_by_users);
+                user.setBlockedByUsers(blockedByUser);
             }
+        }
+    }
+
+    public UserDatabase getDatabase() {
+        return this.database;
+    }
+
+    public void saveNewUser(String username, UserAccount account) {
+        HashMap<String, UserAccount> d = this.database.getData();
+        d.put(username, account);
+    }
+
+    public UserAccount retrieveUserAccount(String username) {
+        return this.database.getData().get(username);
+    }
+
+    public boolean isPairExist(String username, String password) {
+        if (this.database.getData().containsKey(username)) {
+            UserAccount account = retrieveUserAccount(username);
+            return account.getPassword().equals(password);
         } else {
             return false;
         }
     }
 
     public boolean isUsernameExist(String username) {
-        if (this.database.get_data().containsKey(username)) {
-            return true;
-        }
-        return false;
+        return this.database.getData().containsKey(username);
     }
 
-    public int number_user() {
-        return this.database.get_data().size();
+    public int numberUser() {
+        return this.database.getData().size();
     }
 
-    public String[] list_username() {
-        return this.database.get_data().keySet().toArray(new String[0]);
+    public String[] listUsername() {
+        return this.database.getData().keySet().toArray(new String[0]);
     }
 
+    @Override
     public Object[][] writeDatabase() {
         Object[][] userProfile = {{}};
-        String[] usernames = list_username();
-        for (int i = 0; i < number_user(); i++) {
-            UserAccount user = retrieve_user_account(usernames[i]);
-            Object[] userProfileEach = new Object[]{user.get_username(),
-                                                    user.get_full_name(),
-                                                    user.get_age(),
-                                                    user.get_pronouns(),
-                                                    user.get_location().get("Country"),
-                                                    user.get_location().get("Province"),
-                                                    user.get_location().get("City"),
-                                                    user.get_gender(),
-                                                    user.get_sexuality(),
-                                                    user.get_interests(),
-                                                    user.get_password()};
+        String[] usernames = listUsername();
+        for (int i = 0; i < numberUser(); i++) {
+            UserAccount user = retrieveUserAccount(usernames[i]);
+            UserManager userManager = new UserManager(user);
+            Object[] userProfileEach = new Object[]{
```

Comment:
Is there a particular reason you decided to make this an array of objects? It's not obvious to me at a glance.